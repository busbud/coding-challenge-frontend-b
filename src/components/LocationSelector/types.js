// @flow

export type Suggestion = {|
  value: string,
|};
