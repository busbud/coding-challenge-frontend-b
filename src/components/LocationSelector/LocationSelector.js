// @flow
import React from 'react';

import MuiDownshift from 'mui-downshift';
import { withStyles } from '@material-ui/core/styles';
import type { LocationSuggestion } from '../../types';

type Props = {|
  suggestions: Array<LocationSuggestion>,
  label: string,
  onChange: () => Array<any>,
|};

const UnstyledLocationSelector = (props: Props) => {
  const { suggestions, label, onChange } = props;
  return (
    <MuiDownshift
      items={suggestions}
      getInputProps={() => ({
        label,
      })}
      onChange={onChange}
    />
  );
};

export const LocationSelector = withStyles({})(UnstyledLocationSelector);
