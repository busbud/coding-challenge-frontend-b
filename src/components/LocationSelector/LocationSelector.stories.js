// @flow
import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { LocationSelector } from './LocationSelector';

import { suggestions } from './fixtures/arrival-fixture';

storiesOf('LocationSelector', module)
  .add('With Results', () => (
    <LocationSelector suggestions={suggestions} label="label" onChange={action('onChange')} />
  ))
  .add('Without Results', () => (
    <LocationSelector suggestions={[]} label="label" onChange={action('onChange')} />
  ));
