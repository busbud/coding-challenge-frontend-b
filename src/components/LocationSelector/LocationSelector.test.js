import React from 'react';

import renderer from 'react-test-renderer';
import { suggestions } from './fixtures';

import { LocationSelector } from './LocationSelector';

describe('LocationSelector', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<LocationSelector suggestions={suggestions} label="label" onKeyDown={() => {}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
