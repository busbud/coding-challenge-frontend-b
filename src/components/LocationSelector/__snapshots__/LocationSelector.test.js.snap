// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationSelector renders correctly 1`] = `
<div
  aria-expanded={false}
  aria-haspopup="listbox"
  aria-labelledby="downshift-0-label"
  aria-owns={null}
  role="combobox"
>
  <div
    className="MuiFormControl-root-1 MuiFormControl-fullWidth-4"
  >
    <label
      className="MuiFormLabel-root-16 MuiInputLabel-root-5 MuiInputLabel-formControl-10 MuiInputLabel-animated-13"
      data-shrink={false}
      htmlFor="downshift-0-input"
      id="downshift-0-label"
    >
      label
    </label>
    <div
      aria-activedescendant={null}
      aria-autocomplete="list"
      aria-controls={null}
      aria-labelledby="downshift-0-label"
      className="MuiInputBase-root-36 MuiInput-root-23 MuiInput-underline-27 MuiInputBase-formControl-37 MuiInput-formControl-24 MuiInputBase-adornedEnd-41"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoComplete="off"
        className="MuiInputBase-input-46 MuiInput-input-31 MuiInputBase-inputAdornedEnd-52"
        disabled={false}
        id="downshift-0-input"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        required={false}
        type="text"
        value=""
      />
      <div
        className="MuiInputAdornment-root-53 MuiInputAdornment-positionEnd-56"
      >
        <button
          aria-label="Toggle menu open"
          className="MuiButtonBase-root-63 MuiIconButton-root-57"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiIconButton-label-62"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-66"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-76"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
